libcudacxx/test/libcudacxx/cuda/iterators/transform_iterator/member_types.pass.cpp

```diff
@@ -0,0 +1,96 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of libcu++, the C++ Standard Library for your entire system,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#include <cuda/iterator>
+#include <cuda/std/cassert>
+#include <cuda/std/concepts>
+
+#include "test_iterators.h"
+#include "test_macros.h"
+#include "types.h"
+
+template <class Iter, class Fn>
+_CCCL_CONCEPT HasIterCategory =
+  _CCCL_REQUIRES_EXPR((Iter, Fn))(typename(typename cuda::transform_iterator<Iter, Fn>::iterator_category));
+
+__host__ __device__ constexpr bool test()
+{
+  {
+    // Member typedefs for contiguous iterator.
+    static_assert(
+      cuda::std::same_as<cuda::std::iterator_traits<int*>::iterator_concept, cuda::std::contiguous_iterator_tag>);
+    static_assert(
+      cuda::std::same_as<cuda::std::iterator_traits<int*>::iterator_category, cuda::std::random_access_iterator_tag>);
+
+    using TIter = cuda::transform_iterator<int*, Increment>;
+    static_assert(cuda::std::same_as<typename TIter::iterator_concept, cuda::std::random_access_iterator_tag>);
+    static_assert(cuda::std::same_as<typename TIter::iterator_category, cuda::std::random_access_iterator_tag>);
+    static_assert(cuda::std::same_as<typename TIter::value_type, int>);
+    static_assert(cuda::std::same_as<typename TIter::difference_type, cuda::std::ptrdiff_t>);
+  }
+  {
+    // Member typedefs for random access iterator.
+    using TIter = cuda::transform_iterator<random_access_iterator<int*>, Increment>;
+    static_assert(cuda::std::same_as<typename TIter::iterator_concept, cuda::std::random_access_iterator_tag>);
+    static_assert(cuda::std::same_as<typename TIter::iterator_category, cuda::std::random_access_iterator_tag>);
+    static_assert(cuda::std::same_as<typename TIter::value_type, int>);
+    static_assert(cuda::std::same_as<typename TIter::difference_type, cuda::std::ptrdiff_t>);
```

Comment:
Q: Is there any chance we can test `thrust::device_iterator` as well? Because Thrust's iterator categories contain systems.

Reply:
Unfortunately we would need to test that in thrust itself because libcu++ does not depend on thrust.

But yes the system category is something we need to properly work with. However, I believe we need to first decide how we can tag iterators with a system and then transform iterator can just piggy-back on the stored iterator